src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react"
+import { Checkbox, Flex, Select, Spacer } from "@artsy/palette"
+import {
+  useAuctionResultsFilterContext,
+  useCurrentlySelectedFiltersForAuctionResults,
+} from "Apps/Artist/Routes/AuctionResults/AuctionResultsFilterContext"
+import { FilterExpandable } from "Components/ArtworkFilter/ArtworkFilters/FilterExpandable"
+import { ShowMore } from "Components/ArtworkFilter/ArtworkFilters/ShowMore"
+
+export const SaleEndYearFilter: React.FC = () => {
+  const { setFilter, aggregations } = useAuctionResultsFilterContext()
+  const {
+    saleEndYear,
+    saleStartYear,
+    allowUnspecifiedSaleDates,
+  } = useCurrentlySelectedFiltersForAuctionResults()
+
+  const options = (
+    aggregations
+      ?.find(aggregation => aggregation.slice === "LOTS_BY_SALE_YEAR")
+      ?.counts.filter(c => c !== null) || []
+  ).map(c => ({
+    text: c?.name,
+    value: c?.name,
+  }))
+
+  React.useEffect(() => {
```

Comment:
Not sure if this makes sense here, I'm looking for _suggestions_  😄

Reply:
Why can't this logic happen on the select callbacks?

Reply:
Do you have an example? 

```
          onSelect={year =>
                saleStartYear === null && options.length > 0
                  ? setFilter?.("saleStartYear", parseInt(options[0].value))
                  : setFilter?.("saleStartYear", parseInt(year))
              }
```

This won't work as expected :|

Reply:
I suppose I don't follow exactly what this is doing without pulling it; I can take a look in a bit. If the effect works that's fine with me but they are much harder to reason about and frequently introduce bugs when performing something directly in callbacks is more clear.